lib/vrl/compiler/src/enrichment_tables.rs

```diff
@@ -0,0 +1,36 @@
+use dyn_clone::DynClone;
+use std::collections::BTreeMap;
+
+pub trait EnrichmentTables: DynClone {
+    fn get_tables(&self) -> Vec<String>;
+    fn find_table_row(
+        &self,
+        table: &str,
+        criteria: BTreeMap<String, String>,
+    ) -> Result<Option<Vec<String>>, String>;
+    fn add_index(&mut self, table: &str, fields: Vec<&str>) -> Result<(), String>;
+}
+
+dyn_clone::clone_trait_object!(EnrichmentTables);
+
+/// Create a empty enrichment for situations when we don't have any tables loaded.
+#[derive(Clone, Debug)]
+pub struct EmptyEnrichmentTables;
+
+impl EnrichmentTables for EmptyEnrichmentTables {
+    fn get_tables(&self) -> Vec<String> {
+        Vec::new()
+    }
+
+    fn find_table_row(
+        &self,
+        _table: &str,
+        _criteria: BTreeMap<String, String>,
+    ) -> Result<Option<Vec<String>>, String> {
+        Ok(None)
+    }
+
+    fn add_index(&mut self, _table: &str, _fields: Vec<&str>) -> Result<(), String> {
+        Ok(())
+    }
+}
```

Comment:
Could we instead `impl<T: EnrichmentTables> EnrichmentTables for Option<T>`?

Reply:
I probably could, `EmptyEnrichmentTables` is largely just for test code, so I don't really want primary support for it. The main code should always have a populated set of tables (albeit often populated with nothing when the users config doesn't specify any tables).

Reply:
If it's only for tests, we should mark it as such so that it doesn't get compiled into the release code. If it's also used in release code, I think it's more idiomatic to do the impl on `Option`.